src/main/scala/dibl/Fringes.scala

```diff
@@ -0,0 +1,221 @@
+    * TODO merge both sub sets, inner before outer
+    */
+  lazy val reusedLeft: Seq[Link] = intoSide(leftTargetCol) ++ intoSide(leftTargetCol + 1)
+
+  /** The left feet of the `>|` shaped footside stitches alias the pairs coming out of the footside.
+    * TODO merge both sub sets, inner before outer
+    */
+  lazy val reusedRight: Seq[Link] = intoSide(rightTargetCol) ++ intoSide(rightTargetCol - 1)
+
+  /** The red links in the [[svgDoc]],
+    * the pairs needed to start a patch of lace along the top and corners,
+    * each link is one leg of the `v`'s in the ascii art diagram of the class
+    */
+  lazy val newPairs: Seq[Link] = {
+    val row = absSrcNodes(topTargetRow)
+    for {
+      targetCol <- leftTargetCol to rightTargetCol
+      (sourceRow, sourceCol) <- row(targetCol)
+      if fromOutside(targetCol, sourceRow, sourceCol)
+    } yield Link(
+      source = Cell(sourceRow, sourceCol),
+      target = Cell(topTargetRow, targetCol)
+    )
+  }
+
+  private lazy val count = Matrix.countLinks(absSrcNodes)
+
+  private def needsOutLink(outerCol: Int, innerCol: Int): Seq[Cell] = {
+    for {
+      targetRow <- topTargetRow until bottomTargetRow
+      targetCol <- outerCol to(innerCol, innerCol - outerCol)
+      _ <- 1 to (4 - count(targetRow)(targetCol)) % 4
+    } yield Cell(targetRow, targetCol)
+  }
+
+  /** Nodes that need a link going into the left footside.
+    * Nodes that needs two out going links are returned twice.
+    * Nodes in the outer column come before the next column.
+    */
+  lazy val needsOutLeft: Seq[Cell] = needsOutLink(leftTargetCol, leftTargetCol + 1)
+
+  /** Nodes that need a link going into the right footside.
+    * Nodes that needs two out going links are returned twice.
+    * Nodes in the outer column come before the previous column.
+    */
+  lazy val needsOutRight: Seq[Cell] = needsOutLink(rightTargetCol, rightTargetCol - 1)
+
+  /** variants of [[reusedLeft]] and [[reusedRight]] with sources moved
+    * to [[needsOutLeft]] respective [[needsOutRight]]
+    */
+  lazy val footSides: Seq[Link] = {
+
+    val targets = mutable.Stack[Cell]()
+
+    def popLinks(sourceRow: Int, sourceCol: Int): Seq[Link] = for {
+      _ <- 1 to (4 - count(sourceRow)(sourceCol)) % 4 // nr of needed links out
+      if targets.nonEmpty
+    } yield Link(Cell(sourceRow, sourceCol), targets.pop())
+
+    def pushLinks(targetRow: Int, targetCol: Int): Unit = absSrcNodes(targetRow)(targetCol).foreach {
+      case (sourceRow, sourceCol) =>
+        if (fromOutside(targetCol, sourceRow, sourceCol))
+          targets.push(Cell(targetRow, targetCol))
+    }
+
+    def createLinks(cols: Inclusive): IndexedSeq[Link] = {
+      for {
+        row <- bottomTargetRow to (topTargetRow, -1)
+        col <- cols
+        links = popLinks(row, col)
+        _ = pushLinks(row, col)
+      } yield links
+    }.flatten
+
+    createLinks(leftTargetCol to rightTargetCol) ++ createLinks (rightTargetCol to (rightTargetCol - 1, -1))
```

Comment:
The answer by mail landed on the main page of the pull request what makes the discussion hard to follow.

You are right about the draw functions.

Rephrasing the answer to your question: `popLinks` returns another type (nested tuples) and can't be mixed with the two previous expressions (plain integers). Moreover, changing the `=` in `<-` would change the frequency `pushLinks` gets called, it should be called once, regardless of the number of items return by `popLinks`.

Just tested it, made a mistake in the first call to `createLinks` and need to clear the stack between the calls. Otherwise :tada: